### PR TITLE
REFACTOR: Glimmerify topic summarization widgets.

### PR DESCRIPTION
**Pull request by romanrizzi**
Simplifies all the logic for generating/regenerating summaries and expanding/collapsing the summary box. It makes streaming easier to implement since now we can subscribe to message bus directly from the component.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/summary-box.hbs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{{#if @postAttrs.hasTopRepliesSummary}}
+  <p>{{html-safe this.topRepliesSummaryInfo}}</p>
+{{/if}}
+<div class="summarization-buttons">
+  {{#if @postAttrs.summarizable}}
+    {{#if this.canCollapseSummary}}
+      <DButton
+        @class="btn btn-primary topic-strategy-summarization"
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/summary-box.hbs:8]:
```suggestion
        @class="btn-primary topic-strategy-summarization"
```
(already [included in d-button](https://github.com/discourse/discourse/blob/4d2d31e68aa9ba556c6aafac98ebe0158d42b52e/app/assets/javascripts/discourse/app/components/d-button.gjs#L27-L27))

File: app/assets/javascripts/discourse/app/components/summary-box.hbs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        @action={{this.toggleSummary}}
+        @title="summary.buttons.hide"
+        @label="summary.buttons.hide"
+        @icon="chevron-up"
+      />
+    {{else}}
+      <DButton
+        @class="btn btn-primary topic-strategy-summarization"
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/summary-box.hbs:16]:
```suggestion
        @class="btn-primary topic-strategy-summarization"
```
(already [included in d-button](https://github.com/discourse/discourse/blob/4d2d31e68aa9ba556c6aafac98ebe0158d42b52e/app/assets/javascripts/discourse/app/components/d-button.gjs#L27-L27))

File: app/assets/javascripts/discourse/app/components/summary-box.hbs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        @translatedTitle={{this.generateSummaryTitle}}
+        @icon={{this.generateSummaryIcon}}
+        @disabled={{this.loadingSummary}}
+      />
+    {{/if}}
+  {{/if}}
+
+  {{#if @postAttrs.hasTopRepliesSummary}}
+    <DButton
+      @class="btn top-replies"
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/summary-box.hbs:28]:
```suggestion
      @class="top-replies"
```
(already [included in d-button](https://github.com/discourse/discourse/blob/4d2d31e68aa9ba556c6aafac98ebe0158d42b52e/app/assets/javascripts/discourse/app/components/d-button.gjs#L27-L27))

File: app/assets/javascripts/discourse/app/components/summary-box.hbs (hunk 1 above)
Comment (romanrizzi) [reply app/assets/javascripts/discourse/app/components/summary-box.hbs:8]: Will remove it from the second button too.

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**